packages/core/src/external/cda/process-attachments.ts

```diff
@@ -79,26 +88,47 @@ export async function processAttachments({
       mimeType: fileDetails.mimeType,
     });
     const fileUrl = s3Utils.buildFileUrl(s3BucketName, fileKey);
-    const attachment: Attachment = {
-      ...(fileDetails.mimeType && { contentType: fileDetails.mimeType }),
-      url: fileUrl,
-      size: sizeInBytes(fileDetails.fileB64Contents),
-      title: fileKey,
-    };
+
+    const attachment = buildAttachment(fileDetails, fileUrl, fileKey);
 
     if (docRef.date) attachment.creation = docRef.date;
     docRef.content = [{ attachment }];
-    const uploadParams = {
-      bucket: s3BucketName,
-      key: fileKey,
-      file: Buffer.from(fileDetails.fileB64Contents, "base64"),
-      ...(fileDetails.mimeType && { contentType: fileDetails.mimeType }),
-    };
+    const uploadParams = buildUploadParams(fileDetails, s3BucketName, fileKey);
 
     uploadDetails.push(uploadParams);
     docRefs.push(docRef);
   });
 
+  b64Attachments.organizers.map(organizerEntry => {
+    const mediaComponents = getMediaObservations(organizerEntry);
+
+    mediaComponents?.map(mediaEntry => {
+      const obsMedia = mediaEntry.observationMedia;
```

Comment:
like above, could collapse both these maps so that you only do `b64Attachments.obsMedias.map(obsMedia => {...})`

Reply:
for this one, we're adding the whole component, including other observations, so can't really narrow down to obs medias right away. I'll leave this as is

Reply:
I don't see us adding the whole component, you only operate on `obsMedia` from this point onwards no?